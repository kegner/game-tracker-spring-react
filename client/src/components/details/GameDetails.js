import React from 'react'
import { useSelector } from 'react-redux';
import DetailsItem from './DetailsItem';

const GameDetails = () => {

  const { selectedGame } = useSelector(state => state.gameReducer);

  return (
    <div className="card">
      <div className="card-header bg-dark text-white">
        Game Details
      </div>
      <div className="card-body">
        <div className="row mb-3">
          <div className="col-3">
            <DetailsItem label="Title" value={selectedGame.title} />
          </div>
          <div className="col-3">
            <DetailsItem label="Platform" value={selectedGame.platform} />
          </div>
          <div className="col-3">
            <DetailsItem label="Start Date" value={selectedGame.startDate} />
          </div>
          <div className="col-3">
            <DetailsItem label="End Date" value={selectedGame.endDate} />
          </div>
        </div>
        <div className="row mb-3">
          <div className="col-3">
            <DetailsItem label="Status" value={selectedGame.status} />
          </div>
          <div className="col-3">
            <DetailsItem label="Hours to Complete" value={selectedGame.hoursToComplete} />
          </div>
          <div className="col-3">
            <DetailsItem label="Rating" value={selectedGame.rating} />
          </div>
          <div className="col-3">
            <DetailsItem label="Price" value={selectedGame.price} currency={true} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default GameDetails
