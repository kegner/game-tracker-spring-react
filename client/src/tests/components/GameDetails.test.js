import React from 'react';
import { Provider } from "react-redux";
import { render, getByText } from '@testing-library/react';
import { createTestStore } from '../store/testStore';
import GameDetails from '../../components/details/GameDetails';

let store;
let dispatch;
let component;
let container;
let asFragment;

describe("GameDetails", () => {

  beforeEach(() => {
    store = createTestStore({
      mockStore: true,
      initialState: {
        gameReducer: {
          selectedGame: {
            id: "123", title: "My Game"
          }
        }
      }
    });
    dispatch = store.dispatch;
    component = render(
      <Provider store={store}>
        <GameDetails />
      </Provider>
    );
    container = component.container;
    asFragment = component.asFragment;
  });

  it("exists", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("has game title", () => {
    expect(getByText(container, "My Game")).toBeVisible();
  });

});