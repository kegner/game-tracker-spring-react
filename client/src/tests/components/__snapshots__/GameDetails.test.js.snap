// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDetails exists 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-header bg-dark text-white"
    >
      Game Details
    </div>
    <div
      class="card-body"
    >
      <div
        class="row mb-3"
      >
        <div
          class="col-3"
        >
          <strong>
            Title
          </strong>
          <br />
          <span>
            My Game
          </span>
        </div>
        <div
          class="col-3"
        >
          <strong>
            Platform
          </strong>
          <br />
          <span />
        </div>
        <div
          class="col-3"
        >
          <strong>
            Start Date
          </strong>
          <br />
          <span />
        </div>
        <div
          class="col-3"
        >
          <strong>
            End Date
          </strong>
          <br />
          <span />
        </div>
      </div>
      <div
        class="row mb-3"
      >
        <div
          class="col-3"
        >
          <strong>
            Status
          </strong>
          <br />
          <span />
        </div>
        <div
          class="col-3"
        >
          <strong>
            Hours to Complete
          </strong>
          <br />
          <span />
        </div>
        <div
          class="col-3"
        >
          <strong>
            Rating
          </strong>
          <br />
          <span />
        </div>
        <div
          class="col-3"
        >
          <strong>
            Price
          </strong>
          <br />
          <span />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
